### ISSUE TITLE
Sharada with vedic accents results in tofus.

### ISSUE THREAD

**Issue by vvasuki** [googlefonts/noto-fonts#1753]
### Font 
NotoSansSharada-Regular from [hinted directory of this repo](https://github.com/googlefonts/noto-fonts/tree/master/hinted) as of this date.

### OS name and Application
Chrome (Version 81.0.4044.92 (Official Build) (64-bit) on Arch Linux)

### Issue

Open https://vishvasa.github.io/devaH/hindukaH/indraH/Rk/apratiratham/?&transliteration_target=sharada on Chrome version described above vs firefox.

You get bad rendering (Tofus) as shown below:

![image](https://user-images.githubusercontent.com/2664797/82746327-0c07e080-9dac-11ea-9fd8-536e52df5904.png)

### Additional info
- Please note that the problem does not appear on Firefox
- Further, note that analogous problems existed on some older versions of NotoSansKannada-Regular, NotoSansTelugu-Regular, NotoSansMalayalam-Regular, NotoSansTamil-Regular etc.., but were solved when the latest fonts from this repository were used. Also see https://github.com/googlefonts/noto-fonts/issues/1752
- Please see https://github.com/sanskrit-coders/sanskrit-documentation-theme-hugo/issues/7 for related discussion.

**Comment by nizarsq** [comment]
Please include the font details like font source directory, ttf/otf hitnted/unhinted. Thank you.

**Comment by vvasuki** [comment]
Done.

**Comment by simoncozens** [comment]
Unfortunately the website mentioned in this issue is no longer accessible. Could you provide a test string?

If this is still a problem, we will need to decide
* if Vedic accents are in scope for the Sharada font.
* if so, whether new accents need to be designed for Sharada, or if accents from other Noto fonts can be included.

**Comment by vvasuki** [comment]
Fixed the link.

Test string - 

```

𑆄॒𑆯𑆶𑆂 𑆯𑆴𑆯𑆳॑𑆤𑆾(=𑆠𑆵𑆑𑇀𑆰𑇀𑆟𑆾) 𑆮𑆸𑆰॒𑆨𑆾 𑆤 𑆪𑆶॒𑆣𑇀𑆩𑆾(=𑆪𑆾𑆣𑆳)
𑆔॑𑆤𑆳𑆔॒𑆤𑆂(=𑆔𑆳𑆠𑆑𑆂) 𑆑𑇀𑆰𑆾𑆨॑𑆟𑆯𑇀𑆖𑆫𑇀𑆰𑆟𑆵॒𑆤𑆳𑆩𑇀(=𑆥𑆢𑆳𑆠𑆵𑆤𑆳𑆩𑇀)𑇅
𑆱𑆁॒𑆑𑇀𑆫𑆤𑇀𑆢॑𑆤𑆾 𑇁𑆤𑆴𑆩𑆴॒𑆰 𑆍॑𑆑𑆮𑆵॒𑆫𑆂
𑆯॒𑆠ꣳ 𑆱𑆼𑆤𑆳॑ 𑆃𑆘𑆪𑆠𑇀 𑆱𑆳॒𑆑𑆩𑇀 𑆅𑆤𑇀𑆢𑇀𑆫𑆂॑ 𑇆
```

**Comment by simoncozens** [comment]
Confirmed; thank you for the clear test case.

* We are missing the anudatta mark (U+0952). This glyph should definitely be added. 
* We are also missing the candabindu virama (U+A8F3) and similar characters. I am less sure if these extended characters in the Devanagari block should be added to the Sharada font; I feel like they are quite rare but I don't have anything to base that on. I think it would be better for those to be either provided by using a browser font stack which falls back to Noto Sans Devanagari, or by having a merged font containing both Sharada and Extended Devanagari glyphs.

**Comment by vvasuki** [comment]
Samavedic accents too - 

`𑒁꣢𑒏𑓂𑒭꣣𑒢𑓂𑒢𑒧𑒲꣢𑒧𑒠𑒢𑓂𑒞꣣ 𑒯𑓂𑒨𑒫꣢ 𑒣𑓂𑒩𑒱꣣𑒨𑒰 𑒁꣢𑒡𑒴𑒭𑒞। 𑒁𑒮𑓂𑒞𑒼꣢𑒭𑒞꣣ 𑒮𑓂𑒫𑒦𑒰꣢𑒢𑒫𑒼꣣ 𑒫𑒱𑒣𑓂𑒩𑒰꣣ 𑒢𑒫𑒱꣢𑒭𑓂𑒚𑒨𑒰 𑒧꣣𑒞𑒲꣢꣫ 𑒨𑒼𑒖𑒰꣣꣬ 𑒢𑓂𑒫𑒱꣢꣯𑒢𑓂𑒠𑓂𑒩 𑒞𑒹꣣ 𑒯𑒩𑒲꣢ ॥`

And a separate font rendering the text if it helps  - 

![image](https://user-images.githubusercontent.com/2664797/152168828-63e31883-7cf1-468a-a81d-1726751311d0.png)

**Comment by simoncozens** [comment]
That sample text appears to be Tirhuta... But yes, I take your point. The whole range U+A8E0..U+A8F1 are marks that will need to be anchored onto glyphs in this font. I don't know if adding those glyphs is in scope, however.

**Comment by vvasuki** [comment]
(closed by mistake?)

**Comment by marekjez86** [comment]
@vvasuki @simoncozens : it was closed because I merged https://github.com/googlefonts/noto-source/pull/496
I do not think that we ought to include the characters from other scripts, but maybe this will change. I'll check with the uNicode committee what they think

**Comment by vvasuki** [comment]
> @vvasuki @simoncozens : it was closed because I merged [googlefonts/noto-source#496](https://github.com/googlefonts/noto-source/pull/496) I do not think that we ought to include the characters from other scripts, but maybe this will change. I'll check with the uNicode committee what they think

Unicode standards and recommendations are clear:

> U+0951 devanagari stress sign udatta and U+0952 devanagari stress sign anudatta are tone marks used in the representation of Vedic text in Devanagari. These two combining marks **may also occur in the representation of Vedic texts written in other scripts**, including transliterations in the Latin script. They are given the Indic_Syllabic_- Category value of Cantillation_Mark. (Page 476 of https://www.unicode.org/versions/Unicode14.0.0/ch12.pdf)

> Devanagari Extended: U+A8E0–U+A8FF This block of characters is used chiefly for Vedic Sanskrit, although many of the characters are generic and **can be used by other Indic scripts**. (Page 479 of https://www.unicode.org/versions/Unicode14.0.0/ch12.pdf)

> Vedic Extensions: U+1CD0–U+1CFF The Vedic Extensions block includes characters that are used in Vedic texts; they may be used with Devanagari, as **well as many other Indic scripts**. (Page 481 of https://www.unicode.org/versions/Unicode14.0.0/ch12.pdf)

**Comment by marekjez86** [comment]
@vvasuki : thank you for pointing this out. We'll have to start (but we need to get funded first :-)) a project to update Sharada. This is beyond the scope of work that will be done by @simoncozens . I'll reopen this issue.

**Comment by simoncozens** [comment]
I don't know if we do need to update the font further. I've added anudatta, but the other characters are already covered in Noto Sans Devanagari. I'm not convinced that separate Sharada-specific designs are needed. The problem here is one of how we consider fonts which span multiple Unicode scripts. There are a number of options:

* We simply copy the needed glyphs from Noto Sans Devanagari into the Noto Sans Sharada sources or binaries. This is a simple fix, but I would like to think about it more generally. There may be quite a collection of glyphs from other scripts which  should be added to script-specific fonts, so if it is the way we want to proceed, it is something we should automate. (I'm aware, for example, that there are a several outstanding issues with fonts not containing U+25CC DOTTED CIRCLE.) If we do do this, I would prefer to keep the sources script-specific and add the needed glyphs in post-processing.

* Another way to think about it is more holistically, in terms of the whole Noto project and font delivery more generally.  We should be aware that documents often contain more than language and text in more than one script. So it is not really reasonable to expect one script-specific font to be able to render all the characters in a document. (For example, Latin characters and many punctuation characters may be used in certain documents but our script-specific fonts do not include those.) The solution in that case would be either:
  *  a merged Noto subset which contains all the characters "reasonably" expected to render a document primarily in a given script - for example, either a "Noto Sans Sharada+" which contains Devanagari marks, or a "Noto Sans Indic + Latin" which contains a set of glyphs needed to satisfy a range of Indic document processing needs.
  * or perhaps better, encouraging greater use of font fallback stacks - possibly delivering on-the-fly fallback stacks for web use which provide Noto fonts covering all the codepoints used in a document.

Basically, I think this needs thinking through, but not necessarily more design. :-)

**Comment by marekjez86** [comment]
There's a "pull" to merge all Noto scripts into one bundle (but I'm not thrilled about it :-)).
Also, which sounds more reasonable, there were requests to create Noto <whataver-script> + Latin I think by people who download the fonts and want to write in their own language with an occasional use of English. This is similar situation but instead of Latin we could use Devanagari (or Deva + Latin) for Indic languages.

BTW, it has been Noto's motto to "keep the sources script-specific" (a quote from the above comment).
